buildkite/bazelci.py

```diff
@@ -1722,10 +1724,70 @@ def bazel_build_step(
     )
 
 
+def filter_tasks_that_should_be_skipped(task_configs, pipeline_steps):
+    skip_tasks = get_skip_tasks()
+    if not skip_tasks:
+        eprint("No tasks to skip.")
+        return task_configs
+
+    actually_skipped = []
+    skip_tasks = set(skip_tasks)
+    for task in list(task_configs.keys()):
+        if task in skip_tasks:
+            actually_skipped.append(task)
+            del task_configs[task]
+            skip_tasks.remove(task)
+
+    if not task_configs:
+        raise BuildkiteException(
+            "Nothing to do since all tasks in the configuration should be skipped."
+        )
+
+    annotations = []
+    if actually_skipped:
+        annotations.append(
+            ("info", "Skipping the following task(s): {}".format(", ".join(actually_skipped)))
+        )
+
+    if skip_tasks:
+        annotations.append(
+            (
+                "warning",
+                (
+                    "The following tasks should have been skipped, "
+                    "but were not part of the configuration: {}"
+                ).format(", ".join(skip_tasks)),
+            )
+        )
+
+    if annotations:
+        print_skip_task_annotations(annotations, pipeline_steps)
+
+    return task_configs
+
+
+def get_skip_tasks():
+    value = os.getenv(SKIP_TASKS_ENV_VAR, "")
+    return [v for v in value.split(",") if v]
+
+
+def print_skip_task_annotations(annotations, pipeline_steps):
+    commands = [
+        "buildkite-agent annotate --style={} '{}'  --context 'ctx-{}'".format(s, t, hash(t))
+        for s, t in annotations
+    ]
+    pipeline_steps.append(
+        create_step(label=":pipeline: Print information about skipped tasks.", commands=commands)
```

Comment:
nit: I think we usually don't have a '.' at the end of the labels - maybe remove it?